tests/optim/helpers/models.py

```diff
@@ -0,0 +1,29 @@
+from typing import Type
+
+import torch
+
+
+def _check_layer_in_model(
+    self,
+    model: torch.nn.Module,
+    layer: Type[torch.nn.Module],
+) -> None:
+    def check_for_layer_in_model(model, layer) -> bool:

```

Comment:
If you agree with the above suggestion, we don't need another nested inner function here.

```py
def check_layer_in_model(model, layer):
  for _, child in model._modules.items():
    if child is None:
      continue
    if isinstance(child, layer) or check_layer_in_model(child, layer):
      return True
  return False
```

Reply:
Yeah, we can just use the one function with bool outputs to test for the presence and absence of certain layers.